src/org/yccheok/jstock/gui/charting/ChartJDialog.java

```diff
@@ -78,7 +85,7 @@
  *
  * @author  yccheok
  */
-public class ChartJDialog extends javax.swing.JDialog {
+public class ChartJDialog extends javax.swing.JFrame implements WindowListener {
```

Comment:
So far, I don't worry much with the icon, and have no intention to use different icons for main window, and chart windows. My view is, having single icon will help in branding.

There's a shortcoming with tab-interfaced JFrame. User can no longer compare different charts side-by-side.

But so far, the JDialog to JFrame changes is good. I don't hear any good, or bad feedback from users. I think they are pretty OK with the changes...